### ISSUE TITLE
CliRunner does not capture logging output on stdout

### ISSUE THREAD

**Issue by blueyed** [pallets/click#854]
Is this expected?

It feels like a missing feature, so I'd be happy to provide it giving some pointers.

My test:
```py
def test_cli_run_with_args(runner):
    result = runner.invoke(cli.main, ['run', 'printf', '%s', '--'])
    …
```

And in the `run` command I have:

```py
logger.warn('The temporary profile file (%s) has not been created.',
            profile_file_name)
```

Where `logger` is `logger = logging.getLogger('covimerage')`, which gets setup in `__init__.py`:
```py
logger = logging.getLogger('covimerage')
logger.setLevel(logging.INFO)
logger.addHandler(logging.StreamHandler(sys.stdout))
```

When run manually the warning shows up.

It also shows up in pytest's `Captured stdout call`.

**Comment by blueyed** [comment]
That is more a problem of logging using not `sys.stdout` / `sys.stderr` directly, but where it pointed at when the handler gets created, and is therefore kind of expected.

As for me, I am using a handler based on logging's own `_StderrHandler` that will use `sys.stderr` always.

**Comment by kdeldycke** [comment]
This is going to be addressed in v8.4.0 by https://github.com/pallets/click/pull/3391 which introduce a new `capture="fd"` mode to `CliRunner`.
